Sources/NIOFileSystem/Internal/SystemFileHandle.swift

```diff
@@ -1046,6 +1046,70 @@ extension SystemFileHandle: WritableFileHandleProtocol {
             try sendableView._resize(to: size).get()
         }
     }
+
+    public func setTimes(
+        lastAccessTime: FileInfo.Timespec?,
+        lastDataModificationTime: FileInfo.Timespec?
+    ) async throws {
+        try await self.threadPool.runIfActive { [sendableView] in
+            try sendableView._withUnsafeDescriptor { descriptor in
+                if lastAccessTime == nil, lastDataModificationTime == nil {
+                    // If the timespec array is nil, as per the `futimens` docs,
+                    // both the last accessed and last modification times
+                    // will be set to now.
+                    futimens(descriptor.rawValue, nil)
+                } else {
+                    let lastAccessTimespec: timespec
+                    if let lastAccessTime = lastAccessTime {
+                        lastAccessTimespec = timespec(
```

Comment:
Should we just add an extension to `timespec` to init it from `Timespec`? We do it a handful of times so probably worth doing.